### DISCUSSION TITLE
Is it always a good idea to use the nurse command to clean the repository?

### DISCUSSION THREAD

**Discussion by vincenzopalazzo** [coffee-tools/coffee#125] [General]
I think there is a case where the `nurse` command (introduced in the PR https://github.com/coffee-tools/coffee/pull/115) can recover the `storage` status by rescanning the repository and filling in the storage, which can also detect if the storage is in a corrupted state.

With the information contained in the Coffee configuration, we can also check the plugins that the user has installed.

Therefore, I believe we can explore the idea of making the `coffee nurse` command smarter, if it's worth the effort.

**Comment by tareknaser** [comment]
This sounds interesting and can help speed up the process of developing plugins.
the developer can just drag and drop the repository of the plugin and then run coffee nurse so that coffee detects the new plugin and adds it to the list of plugins that can be installed.

I also wanted to discuss the possible usage of the `nurse` command because while I am working on it, I think of many situations of corruptions that can happen.
For now, `coffee nurse`:
- deletes any unrelated files or folders in the `repositories` folder
- checks if the existing local repositories clones don't exist, if so remove them from configuration and removes the installed plugins 

but I can easily think of corner cases that are not handled.
what if a user deletes a folder inside the repository? (`coffee nurse` doesn't check for all plugins inside all repositories. it  just checks if the parent folder exists)
we can solve this by just iterating over every plugin in the repositories folder
but again, what if a user deletes a single file inside a plugin folder? now the plugin can't run properly but we have no way to know except by using git.

**Reply by tareknaser** [reply]
Of course, the `remote_sync` function should take into account all possible conditions of a corrupt status including if the `exec_path` of the plugin doesn't exist.
This is also related to #122 

I think we should have a clear list of expected situations that result in a corrupt state, then check for them in `remote_sync` and solve the issues with `coffee nurse` command.

**Reply by vincenzopalazzo** [reply]
>but again, what if a user deletes a single file inside a plugin folder? now the plugin can't run properly but we have no way to know except by using git.

This is why we use git :) git support all for us so we should just see how we can extract all this information.

>I think we should have a clear list of expected situations that result in a corrupt state, then check for them in remote_sync and solve the issues with coffee nurse command.

Does this sound like a good tracking issue?

**Reply by tareknaser** [reply]
I opened a new issue #128 to track the various possible functions of the `nurse` command. 

However, I think that we can get it merged when it's ready to solve the first 2 issues in the list and then comeback to it when we finalize the update command #43
